lxd/apparmor/archive.go

```diff
@@ -112,6 +112,16 @@ func archiveProfile(outputPath string, allowedCommandPaths []string) (string, er
 		imagesPath = imagesPathFull
 	}
 
+	derefCommandPaths := make([]string, len(allowedCommandPaths))
+	for i, cmd := range allowedCommandPaths {
+		cmdFull, err := filepath.EvalSymlinks(cmd)
```

Comment:
Is this run in the host context? what if the symlink points to something outside of the container?

Also, is `EvalSymlinks` recursive? If so, what if the path is absolute and the destination does exist on the host but is another symlink?

Reply:
EvalSymlinks is recursive. The extraction command is run in the host context.

Reply:
That's what we use just about everywhere else in our apparmor profiles and that's always lined up with apparmor's own deref behavior. For some reason the archive unpacker logic didn't use it for the extra set of commands (compressors/decompressors), leading to this issue on systems where /bin is a symlink to /usr/bin.

Reply:
does that mean that the problems I described exist but are currently ignored both here and in other places(possibly even by the kernel)?
That doesn't sound good :thinking: